staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch.go

```diff
@@ -718,80 +719,131 @@ func diffListsOfMaps(original, modified []interface{}, schema LookupPatchMeta, m
 	patch := make([]interface{}, 0, len(modified))
 	deletionList := make([]interface{}, 0, len(original))
 
-	originalSorted, err := sortMergeListsByNameArray(original, schema, mergeKey, false)
+	originalKeys, originalMap, err := mergeListToMapAndKeys(original, schema, mergeKey)
 	if err != nil {
 		return nil, nil, err
 	}
-	modifiedSorted, err := sortMergeListsByNameArray(modified, schema, mergeKey, false)
+	modifiedKeys, modifiedMap, err := mergeListToMapAndKeys(modified, schema, mergeKey)
 	if err != nil {
 		return nil, nil, err
 	}
 
-	originalIndex, modifiedIndex := 0, 0
-	for {
```

Comment:
The structure of this change makes it really hard to reason about the side effects, and there's no reasonable way to mitigate a bug introduced by this change.

We have to be 100% sure this won't regress stuff that currently is working, and have a controlled way to roll back a change to the old behavior if an issue is found. Normally this takes the shape of an isolated code change guarded by a feature gate that can be opted into to start, then opted out of as a mitigation.

Reply:
May you give any references on how this usually done? I guess i could isolate the changes

Reply:
Normally this is done with a feature gate, and an isolated code path only active when the gate is enabled, but if the behavior change is client-side, that's harder to gate.